src/main/java/com/networknt/schema/AnyOfValidator.java

```diff
@@ -106,6 +118,14 @@ public Set<ValidationMessage> validate(JsonNode node, JsonNode rootNode, String
                 allErrors.addAll(errors);
             }
 
+            // determine only those errors which are NOT of type "required" property missing
+            Set<ValidationMessage> childNotRequiredErrors = allErrors.stream().filter(error -> !ValidatorTypeCode.REQUIRED.getValue().equals(error.getType())).collect(Collectors.toSet());
+
+            // in case we had at least one (anyOf, i.e. any number >= 1 of) valid subschemas, we can remove all other errors about "required" properties
+            if (numberOfValidSubSchemas >= 1 && childNotRequiredErrors.isEmpty()) {

```

Comment:
Are you sure this condition is correct? If anyOf is valid (so one or more parts are present) then shouldn't we always disregard required errors? Now it seems like you're only disregarding them only if there are no other errors

Reply:
My idea with this was that in case there are other errors as well, for an anyOf scenario, you'll never know which of the alternatives have been provided by intention, and which have been provided by incident, so maybe we should not fiddle with the "required" error messages.

(Note that this is different to oneOf, as in the "oneOf" case it is much more likely that the single valid subschema alternative is indeed the intended one and not valid by incident.)

But I see your point and agree that one/we can also argue to leave the second part of the condition () out...

@stevehu : Which path would you like to go?